docs/architecture.md

```diff
@@ -1,188 +1,260 @@
+specifies, right?-->
 
-`policy-server` performs the validation of policies's settings by
-invoking the `validate_setting` function exposed by each policy.
-This topic is covered more in depth inside
-of [this section](/writing-policies/spec/01-intro-spec.md) of the documentation.
+At start time, the `policy-server` reads its configuration
+and downloads all the Kubewarden policies specified.
+You can download policies from remote HTTP servers and container registries.
 
-`policy-server` will exit with an error if one or more policies received wrong
-configuration parameters from the end user.
+You use policy configuration parameters to tune a policies' behavior.
+After startup and policy download the `policy-server`
+checks the policy settings provided by the user are valid.
 
-If all the policies are properly configured, `policy-server` will spawn a
-pool of worker threads to evaluate incoming requests using the Kubewarden
-policies specified by the user.
+The `policy-server` validates policy settings by invoking
+the `validate_setting` function exposed by each policy.
+There is further documentation in the
+[writing policies section](/writing-policies/spec/01-intro-spec.md)
+of the documentation.
 
-Finally, `policy-server` will start a HTTPS server that listens to incoming
-validation requests. The web server is secured using the TLS key and certificate
-that have been previously created by `kubewarden-controller`.
+<!--TODO:Next paragraph. 'end-user'? How are the configuration parameters
+specified for the `policy-server`-->
 
-Each policy is exposed by the web server via a dedicated path that follows this
-naming convention: `/validate/<policy ID>`.
+The `policy-server` exits with an error if one or more policies
+received wrong configuration parameters from the end-user.
 
-This is how the cluster looks like once the initialization of `policy-server`
-is completed:
+<!--TODO:Can we show the pool of worker threads in the diagram? Or maybe
+it's the stacked things?-->
+
+If all policies are correctly configured,
+`policy-server`
+spawns a pool of worker threads
+to evaluate incoming requests
+using the Kubewarden policies
+specified by the user.
+
+<!--TODO:Is the HTTPS server referred to, that which is in the aquamarine
+box, the 'Kubewarden Policy Server-->
+
+Finally, the `policy-server` starts a HTTPS server
+listening to incoming validation requests.
+Kubewarden uses the TLS key and certificate
+created by `kubewarden-controller`
+to secure the web server.
+
+The web server exposes each policy by a dedicated path
+following the naming convention: `/validate/<policy ID>`.
+
+This diagram shows the cluster when initialization of `policy-server` is complete:
 
 ![policy-server initialized](/img/architecture_sequence_03.png)
 
 ### Making Kubernetes aware of the policy
 
+<!--TODO:Next paragraph. Says 'admission reviews', should that maybe be
+'admission requests'-->
```

Comment:
Not really. The resource kind is actually [`AdmissionReview`](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#webhook-request-and-response).

Reply:
OK, so AdmissionReview is a procedure happening when processing Admission Requests?. I think I need some more explanation for this.

Reply:
Ok thanks using `AdmissionReview` and a link